datafusion/expr/src/udf.rs

```diff
@@ -490,6 +547,40 @@ pub trait ScalarUDFImpl: Debug + Send + Sync {
         self.return_type(arg_types)
     }
 
+    /// What [`DataType`] will be returned by this function, given the
```

Comment:
I also recommend removing the comments from `return_type_from_exprs` now that it is deprecated and you have moved the content here